src/components/Gauge.tsx

```diff
@@ -0,0 +1,216 @@
+	}
+
+	.gauge__legend-item {
+		display: flex;
+		padding: 3px 0 3px 2px;
+		line-height: 9px;
+		width: 50%;
+		float: left;
+	}
+
+	.gauge__legend-color {
+		display: inline-block;
+		flex: 0 0 auto;
+		width: 10px;
+		height: 10px;
+		margin-right: 4px;
+	}
+
+	.gauge__legend-name {
+		font-size: 11px;
+	}
+`;
+
+class StatusGauge extends React.Component<GaugeProps, {}> {
+	private doughnutChart: Chart | null;
+	private doughnutCtx: CanvasRenderingContext2D | null;
+
+	constructor(props: GaugeProps) {
+		super(props);
+	}
+
+	componentDidMount() {
+		this.generateDoughnut(this.props);
+	}
+
+	componentWillReceiveProps(nextProps: GaugeProps) {
+		this.generateDoughnut(nextProps);
+	}
+
+	componentWillUnmount() {
+		this.destroyChart();
+	}
+
+	generateDoughnut(props: GaugeProps) {
+		if (!this.doughnutCtx) {
+			return false;
+		}
+		// Redraw the chart on props update
+		this.destroyChart();
+
+		const ctx = this.doughnutCtx;
+
+		const { title } = props;
+		const { values, colors, labels, count } = this.normalizeProps(props.data);
+
+		const data = this.getGraphData(values, colors, labels, count);
+		const options = this.getGraphOptions(count, title);
+
+		this.doughnutChart = new Chart(ctx, {
+			type: 'doughnut',
+			data,
+			options,
+			plugins: [doughnutLabel],
+		});
+	}
+
+	normalizeProps(data: ChartEntry[]) {
+		const values = data.map(x => x.value);
+		const labels = data.map(x => x.name);
+		const colors = data.map(x => x.color);
+		const count = values.reduce(
+			(carry: number, item: number) => carry + item,
+			0,
+		);
+
+		return { values, labels, colors, count };
+	}
+
+	getGraphData(
+		values: number[],
+		colors: string[],
+		labels: string[],
+		count: number,
+	) {
+		return {
+			labels,
+			datasets: [
+				{
+					backgroundColor: count > 0 ? colors : this.props.placeholderColor,
+					data: count > 0 ? values : [1],
+				},
+			],
+		};
+	}
+
+	getGraphOptions(count: number, title: string) {
+		const pluginOptions = this.getLabelOptions(count, title);
+
+		return {
+			responsive: true,
+			maintainAspectRatio: false,
+			legend: {
+				display: false,
+			},
+			elements: {
+				arc: {
+					borderWidth: 0,
+				},
+			},
+			tooltips: { enabled: count > 0 },
+			plugins: { doughnutLabel: pluginOptions },
+		};
+	}
+
+	getLabelOptions(count: number, title: string) {
+		return {
+			primaryLabel: {
+				...doughnutDefaults.primaryLabel,
+				font,
+				size: count > 999 ? 22 : count > 99 ? 30 : 38,
+				offsetFromCenter: count > 999 ? 8 : 12,
+				text: count,
+			},
+			secondaryLabel: {
+				...doughnutDefaults.secondaryLabel,
+				font,
+				text: title.toUpperCase(),
+			},
+		};
+	}
+
+	destroyChart() {
+		if (this.doughnutChart) {
+			this.doughnutChart.destroy();
+			this.doughnutChart = null;
+		}
+	}
+
+	render() {
+		return (
+			<StatusGaugeWrapper {...this.props}>
+				<div className="gauge__graph">
+					<canvas
+						ref={canv =>
+							(this.doughnutCtx = canv ? canv.getContext('2d') : null)
```

Comment:
This is really weird, you assign `canv` to `this.doughnutCtx` and then immediately overwrite it I think. Should the `=` here be `===` ?
Either way I'd prefer to see the private property just be the Canvas, and have the `getContext` call happen on demand in `generateDoughnut`.

Reply:
Yeah, completely overlooked this one. Will sort it out.